docs/manual/other-topics/upgrade-to-v6.md

```diff
@@ -46,6 +46,51 @@ This method now tests for equality with [`_.isEqual`](https://lodash.com/docs/4.
   await instance.save(); // will save
 ```
 
+### Refactored hooks
+
+In order to streamline API:
+
+- All method style add hook functions have been removed in favor of a composition based approach.
+- Hook names have been removed, you can add and remove them by function reference instead which was supported before.
+- Another notable change that `this` inside of hooks no longer refers to the the the hook subject, it should not be used.
```

Comment:
This is a pretty big breaking change imo, this is very useful writing common hook functions.

Reply:
The primary motivation is to avoid having `this` change in context, you know where that usually leads to.
I wonder if it would be better to explicitly pass the model (class) as the first parameter for every hook.

Reply:
> The primary motivation is to avoid having this change in context, you know where that usually leads to.

`this` is not bad, it is useful depends on how it is used, just like every other feature. I would like to keep it.

>  I wonder if it would be better to explicitly pass the model (class) as the first parameter for every hook.

That introduces another breaking change, I am ok with cleaning API by moving hooks to `something.hooks` but other things must be kept same